### ISSUE TITLE
--new-window not working

### ISSUE THREAD

**Issue by OmgImAlexis** [atom/atom#13526]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

--new-window doesn't work at all

### Steps to Reproduce

1. `atom . --new-window`
2. `atom ../another-dir --new-window`

**Expected behavior:** [What you expect to happen]
Both to open in new WINDOWS

**Actual behavior:** [What actually happens]
Both open in the same window but different panes/tabs

**Reproduces how often:** [What percentage of the time does it reproduce?]
100% of the time.

### Versions

```
Atom: 1.12.7
Electron: 1.3.13
Chrome: 52.0.2743.82
Node: 6.5.0
apm: 1.12.9
npm: 3.10.5
node: 4.4.5
python: 2.7.13
git: 2.11.0
OS: Mac 10.12.1 (16B2659)
```

### Additional Information

I noticed before running `atom --clear-window-state` it was doing the same where it'd open in the same window but different pane but was also causing that window to go from Mac's full screen mode into normal windowed mode as if it was opening a new window.

**Comment by 50Wliu** [comment]
I can't reproduce this on Windows 10 x64, Atom 1.14.0-dev-3e33aa1.

**Comment by GeekAb** [comment]
Not able to reproduce it with : 

Atom: 1.12.7
Electron: 1.3.13
Chrome: 52.0.2743.82
Node: 6.5.0
OS : Mac 10.11.6

**Comment by OmgImAlexis** [comment]
Video showing without safe mode. https://vid.me/l3VT
Video showing with safe mode. https://vid.me/DPaR

From some more tests it seems to mainly do it once the first window is in OSX's full screen mode.

**Comment by OmgImAlexis** [comment]
@50Wliu is there anything else I can provide?

**Comment by ungb** [comment]
@OmgImAlexis does this work when you aren't in full screen mode?

**Comment by ungb** [comment]
@OmgImAlexis looking at your videos, looks like it works in safe mode but doesn't without.  It could be related to a package you have installed.  Could be related to personal/community package you have installed.  

Can you provide a list of packages you have installed? (you can find this by running `apm list` and looking under the section that says `Community Packages`)

**Comment by OmgImAlexis** [comment]
Actually it shows the bug with both safe mode and not but only if you're in OSX's full screen mode. I did more tests and that seems to be the trigger. This was happening after a clean install of OSX before I installed any Atom plugins.

```
Community Packages (13) /Users/xo/.atom/packages
├── autohide-tree-view@0.25.1
├── color-picker@2.2.3
├── file-icons@1.7.25
├── git-status@0.3.4
├── language-docker@1.1.7
├── language-mako@0.6.0
├── language-pug@0.0.19
├── language-vue@0.21.0
├── linter@1.11.18
├── linter-swagger@0.3.8
├── seti-syntax@1.0.1
├── seti-ui@1.4.0
└── tree-view-git-status@1.2.0

└── (empty)
```

**Comment by OmgImAlexis** [comment]
When not in fullscreen mode it seems to behave correctly.

**Comment by jrs65** [comment]
I'm also having the same issue on MacOS 10.12.2. When I have an atom window already in full screen mode, attempting to open another from the command line with `atom -n` just open into the existing window. Out of full screen mode, the command works as expected.

**Comment by 50Wliu** [comment]
Still cannot reproduce on Windows, assuming macOS-only.

**Comment by jrs65** [comment]
A few quick updates, you get exactly the same issue if you just have a window already full screen, and then go to the menubar to do `File -> New Window`.

Also what it does is add a new tab into the existing window (rather than creating a whole new window), but it's not an editor style tab (it's decorated like a standard MacOS tab). I've attached a screenshot taken in safe mode.

<img width="1680" alt="screen shot 2017-01-06 at 10 20 28" src="https://cloud.githubusercontent.com/assets/695206/21728280/323cd6fe-d3fa-11e6-9f52-f47491a6f0ef.png">

**Comment by lee-dohm** [comment]
If @jrs65's screenshot is what ends up happening, this is the expected behavior on the newest versions of macOS. While these are visualized as "tabs", they're OS-level window tabs, not Atom tabs. This only happens in full screen mode and is the expected behavior in full screen mode. Since this is the expected behavior of macOS, I'm going to close this.

If I'm misunderstanding things, please let me know and I'll be happy to reopen the Issue.

**Comment by OmgImAlexis** [comment]
@lee-dohm this may be the expected behaviour of the OS but isn't the expected behaviour of the program and there's still the issue of it also causing the OSX full screen window to minimise down to a normal window at times.

`-n` should open a new window and in all OSs this works the same apart from macOS. I don't see how that's invalid since most of people would expect it to be the exact same as other OSs.

**Comment by 50Wliu** [comment]
I'm pretty sure this is because macOS "helpfully" intercepts the new window command and creates an OS tab for it instead.  This looks like #12690, which is blocked on electron/electron#6124.

**Comment by OmgImAlexis** [comment]
@50Wliu would you be willing to reopen this on the condition it's closed once the other issues are addressed?

**Comment by lee-dohm** [comment]
@OmgImAlexis No, this is a duplicate of #12690, there's no point in having two issues open for the same thing.
